### ISSUE TITLE
twisted.web.wsgi incorrectly unquotes QUERY_STRING

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4143]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)| @therve reported|
|-|-|
|Trac ID|trac#4143|
|Type|defect|
|Created|2009-12-01 17:05:29Z|
|Branch|https://github.com/twisted/twisted/tree/wsgi-query-string-4143|

In `_WSGIResponse.__init__`, the data put in the `QUERY_STRING` env value is passed through urllib.unquote. Although the `WSIG` spec is unclear about it, it looks like the query string should be passed unchanged, and lets the application deals with it.

<details><summary>Searchable metadata</summary>

```
trac-id__4143 4143
type__defect defect
reporter__therve therve
priority__normal normal
milestone__Twisted_10_0 Twisted-10.0
branch__branches_wsgi_query_string_4143 branches/wsgi-query-string-4143
branch_author__therve therve
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1259687129000000 1259687129000000
changetime__1266767176000000 1266767176000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve commented|
|-|-|

(In [28255]) Branching to 'wsgi-query-string-4143'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve removed owner|
|-|-|

This is ready to review in the attached branch.

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|spiv set owner to @therve|
|-|-|

A quick skim of `wsgiref` suggests you're right that it should be unquoted.  In that case, this change looks good to me.  So, assuming the rest of the test suite passes, all that is missing is a news file.  With a news file this will be good to merge.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set status to `closed`|
|-|-|

(In [28295]) Merge wsgi-query-string-4143

Author: therve
Reviewer: spiv
Fixes: [#4143](https://github.com/twisted/twisted/issues/4143)

Remove the call to unquote on QUERY_STRING in twisted.web.wsgi, to pass the
string unchanged to the application layer. It seems to be the expected behavior
of WSGI containers.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
